tests/cpp_check.sh

```diff
@@ -2,10 +2,19 @@
 
 set -e
 
-u="$u -U __REDIRECT_NTH -U _BSD_RUNE_T_ -U _TYPE_size_t -U __LDBL_REDIR1_DECL"
-supp="--suppress=missingInclude --suppress=unusedFunction"
-suppf="redblack.c"
-arg="-q --force $u --enable=all $inc $supp --error-exitcode=1"
-cppcheck="xargs cppcheck $arg"
-
-find . -name "*.[c|h]" | grep -v "$suppf" | $cppcheck  --language=c++ 2>&1
+if [ -n "$CI" ] ; then
+  sudo apt install -y gcc
+  sudo apt install -y make
+  sudo apt install -y autoconf
+  sudo apt install -y automake
+  sudo apt install -y cppcheck
+fi
+ 
+if command -v cppcheck > /dev/null; then
+  u="$u -U __REDIRECT_NTH -U _BSD_RUNE_T_ -U _TYPE_size_t -U __LDBL_REDIR1_DECL"
+  supp="--suppress=missingInclude --suppress=unusedFunction"
+  suppf="redblack.c"
```

Comment:
This line can be removed, this file only exists in inotify-tools and the `| grep -v "$suppf"` part.

Reply:
@ericcurtin 
If you look at this test result, i would like to reduce cppchecks warnings
It is accepted ? 
Would you prefer separate PR?

Reply:
I am happy to merge like this or wait until green. We certainly have to go through them and fix what's worthwhile, this warning can be turned off "noExplicitConstructor", that's one that's not worth fixing.

Reply:
Once this build is green we will stop merging things that are not green